src/Build/Construction/ProjectElementContainer.cs

```diff
@@ -64,32 +65,41 @@ internal ProjectElementContainer(XmlElement xmlElement, ProjectElementContainer
         /// </summary>
         public IEnumerable<ProjectElement> AllChildren => GetChildrenRecursively();
 
+        internal IEnumerable<T> GetAllChildrenOfType<T>()
+            where T : ProjectElement
+            => GetChildrenRecursivelyOfType<T>();
+
         /// <summary>
         /// Get enumerable over all the children
         /// </summary>
         public ICollection<ProjectElement> Children
         {
             [DebuggerStepThrough]
-            get
-            {
-                return new Collections.ReadOnlyCollection<ProjectElement>(
-                        new ProjectElementSiblingEnumerable(FirstChild));
-            }
+            get => new Collections.ReadOnlyCollection<ProjectElement>(new ProjectElementSiblingEnumerable(FirstChild));
```

Comment:
How common would it be that there are no children? In such cases it would be possible to return a singleton empty immutable collection here.